cmd/object-handlers-common.go

```diff
@@ -252,6 +253,20 @@ func isETagEqual(left, right string) bool {
 	return canonicalizeETag(left) == canonicalizeETag(right)
 }
 
+// setAmzExpirationHeader sets x-amz-expiration header with expiry time
+// after analyzing the current bucket lifecycle rules if any.
+func setAmzExpirationHeader(w http.ResponseWriter, bucket string, objInfo ObjectInfo) {
+	if lc, err := globalLifecycleSys.Get(bucket); err == nil {
+		ruleID, expiryTime := lc.PredictExpiryTime(objInfo.Name, objInfo.UserTags)
+		if !expiryTime.IsZero() {
+			w.Header().Set(
+				xhttp.AmzExpiration,
```

Comment:
Rather than using Set() to preserve the case we can do 

```
w.Header()[xhttp.AmzExpiration] = []string{fmt.Sprintf(`expiry-date="%s", rule-id="%s"`, expiryTime.Format(http.TimeFormat)}
```

Would be better to allow all applications to work properly, some AWS SDKs are pretty bad in terms of handling of the HTTP headers, preserving the case allows us to work without issues :-).